CMakeLists.txt

```diff
@@ -870,22 +871,23 @@ if (ENABLE_FUZZING)
   add_subdirectory(fuzz)
 endif ()
 
+# add headers sanity check target that includes all headers independently	
+set(check_headers_dir "${PROJECT_BINARY_DIR}/check-headers")	
+file(GLOB_RECURSE headers_to_check	
+  ${PROJECT_BINARY_DIR}/*.hpp	
+  ${PROJECT_SOURCE_DIR}/include/*.hpp)	
+foreach(header ${headers_to_check})	
+  if ("${header}" MATCHES ".*/include/nodejs/.*")
+    # we do not check NodeJS bindings headers
```

Comment:
NodeJS bindings contain `nan.h`, i.e. we would have to add NAN as a dependency to `check-headers` target.